qa/suites/fs/mgr-failover/mgr-failover.yaml

```diff
@@ -0,0 +1,170 @@
+overrides:
+  ceph:
+    log-ignorelist:
+      - failed to encode map
+  conf:
+    mon:
+      mon warn on legacy crush tunables: false
+    client:
+      client.backup:
+        admin socket: /var/run/ceph/cephfs-backup.asok
+        pid file: /var/run/ceph/cephfs-backup.pid
+roles:
+  - - mon.a
+    - mgr.x
+    - mgr.z
+    - mds.a
+    - osd.0
+    - osd.1
+    - client.0
+    - mon.b
+    - mgr.y
+    - osd.2
+    - osd.3
+    - mds.b
+    - client.1
+    - client.backup
+run-snap-schedule:
+  - exec:
+      mon.a:
+        - "ceph mgr module enable snap_schedule"
+        - "sleep 2"
+        - "ceph config set mgr mgr/snap_schedule/allow_m_granularity True"
+        - "sleep 2"
+        - "ceph fs snap-schedule add /client.0/snap_schedule_test_dir 1M"
+        - "sleep 1140"
+        - "ceph fs snap-schedule deactivate /client.0/snap_schedule_test_dir 1M"
+        - "sleep 60"
+run-cephfs-mirroring:
+  - exec:
+      mon.a:
+        - "ceph mgr module enable mirroring"
+        - sleep 5
+        - "ceph fs volume create backupfs"
+        - sleep 5
+        - "ceph fs snapshot mirror enable cephfs"
+        - sleep 5
+        - "ceph fs snapshot mirror peer_bootstrap create backupfs client.backup_remote remote-site > bootstrap.txt"
+        - sleep 5
+        - "ceph fs snapshot mirror peer_bootstrap import cephfs $(cat bootstrap.txt | jq .token)"
+        - "rm bootstrap.txt"
+        - sleep 5
+        - "ceph fs snapshot mirror add cephfs /client.0/snap_schedule_test_dir"
+        - sleep 1170
+        - "ceph mgr module disable mirroring"
+        - sleep 5
+  - exec:
+      client.1:
+        - "mkdir -p $TESTDIR/mnt.1"
+        - "ceph-fuse --id 1 --client_fs backupfs $TESTDIR/mnt.1"
+run-mgr-failover:
+  - exec:
+      mon.a:
+        # thrash mgr
+        - sleeps=( 60 30 30 30 30 60 30 30 30 60 120 );
+          sleep 300;
+          for i in ${sleeps[*]}; do
+            sleep $i;
+            active_mgr=$(ceph mgr dump | jq --raw-output .active_name);
+            echo "killing active mgr '$active_mgr'";
+            ceph mgr fail $active_mgr;
```

Comment:
Any reason to not have this as a thrasher task?

Reply:
- there wasn't anything significantly sophisticated in the thrasher that I wrote up
- there was a single file that contained everything needed to address the effort

Reply:
What do you mean? If you look at mds_thrash.py, there are a handful of tunables that are used to thrash the mds if different ways/time-points. Not all of those tunables are valid to ceph-mgr, however, I think a thrasher task is much suited for this.